src/components/inputs/money-input/money-input.js

```diff
@@ -502,202 +336,369 @@ class MoneyInput extends React.Component {
         const fakeEvent = {
           persist: () => {},
           target: {
-            id: MoneyInput.getAmountInputId(this.props.id),
-            name: getAmountInputName(this.props.name),
+            id: MoneyInput.getAmountInputId(props.id),
+            name: getAmountInputName(props.name),
             value: formattedAmount,
           },
         };
-        this.props.onChange(fakeEvent);
+        props.onChange(fakeEvent);
       }
     }
-  };
+  }, [props.id, props.name, props.value.amount]);
 
-  containerRef = React.createRef();
-  amountInputRef = React.createRef();
-
-  render() {
-    const hasNoCurrencies = this.props.currencies.length === 0;
-    const hasFocus = this.state.currencyHasFocus || this.state.amountHasFocus;
-
-    const currencySelectStyles = createCurrencySelectStyles({
-      hasWarning: this.props.hasWarning,
-      hasError: this.props.hasError,
-      isDisabled: this.props.isDisabled,
-      isReadOnly: this.props.isReadOnly,
-      hasFocus,
-      menuPortalZIndex: this.props.menuPortalZIndex,
-    });
-    const options = this.props.currencies.map(currencyCode => ({
-      label: currencyCode,
-      value: currencyCode,
-    }));
-
-    const option = (() => {
-      const matchedOption = options.find(
-        optionCandidate =>
-          optionCandidate.value === this.props.value.currencyCode
-      );
-      if (matchedOption) return matchedOption;
-      // ensure an option is found, even when the currencies don't include
-      // the money value's currencyCode
-      if (this.props.value.currencyCode.trim() !== '')
-        return {
-          label: this.props.value.currencyCode,
-          value: this.props.value.currencyCode,
+  const handleAmountChange = React.useCallback(
+    event => {
+      if (isNumberish(event.target.value)) {
+        props.onChange({
+          persist: () => {},
+          target: {
+            id: MoneyInput.getAmountInputId(props.id),
+            name: getAmountInputName(props.name),
+            value: event.target.value,
+          },
+        });
+      }
+    },
```

Comment:
I feel so stupid for still not getting why onChange aint a dep but some part of me gets it.